notebooks/simone_pipeline.md

```diff
@@ -0,0 +1,134 @@
+- Next, `counting.filtering_and_counting` is called
+
+### Arguments:
+
+List Simone's, they're well documented. 
+
+```
+fpath_img_to_filter: str
+path to the file to process
+filtered_png_img_gene_dirs: list
+list of the paths of the directories where the filtered images as are
+saved as pngs.
+filtered_img_gene_dirs: list
+list of the paths of the directories where the filtered images are saved
+as .npy.
+counting_gene_dirs: list
+list of the paths of the directories where the countings of the filtered
+images are saved.
+illumination_correction: bool
+if True the illumination correction is run on the dataset.
+plane_keep: list
+start and end point of the z-planes to keep. Default None
+keep all the planes (ex. [2,-3]).
+min_distance: int
+minimum distance between dots. (peaks)
+stringency: int
+stringency use to select the threshold used for counting. (default 0)
+skip_genes_counting: list
+list of the genes to skip for counting count.
+skip_tags_counting: list
+list of the tags inside the genes/stainings name to avoid to count.
+```
+
+Procedure: 
+
+Note: Expects np.float64 images (how to address this? will everything work?)
+
+1. `filtering.nuclei_filtering`
+	1. called if this is in part of skip-genes, designed to skip nuclei, I think
+  2. 3d gaussian filter, sigma (2, 100, 100) (z, y, x) (could be z, x, y), but x <-> y for our purposes.
+  3. set negative values to zero
+  4. max project over z (np.amax(stack, axis=0)
+2. otherwise, call `filtering.smFISH_filtering`
+  1. `gaussian_filter`, sigma=(1, 8, 8)
+  2. `gaussian_laplace(stack, (0.2, 0.5, 0.5)`  # todo look up
+    3. this makes the peaks negative, so invert signal `img_stack = -img_stack`
+    4. ... and zero out the negative values again
+    5. max project over z `(np.amax(img_stack, axis=0)`
+3. `dots_calling.thr_calculator`
+  1. Counts dots across the whole image using `img_filtered` results, `min_distance`, and `stringency`
+
+### Thr Calculator
+
+This method calculates a threshold to determine if dots are peaks or not. Specifically, it estimates
+threshold_abs for `skimage.feature.peak_local_max`
+
+Arguments:
+1. `min_distance`: if two peaks are not > 5 pixels apart they are the same peak
+2. stringency: higher stringency == higher threshold from `thr_array` to get peaks (tuning parameter)
+
+Returns:
+
+Lots of stuff in the form of a dictionary
+1. `selected_thr`: threshold for counting after application of stringency
+2. `calculated_thr`: threshold after all the trimming is calculated but before stringency application
+3. `selected_peaks`: 2d coordinates of the peaks
+4. `thr_array`: array of 100 points between Img.min() and Img.max(); this is a linspace between min and max
+5. `peaks_coords`: coordinates of peaks at each threshold value.
+6. `total peaks`: number of peaks at each threshold
+7. `thr_idx`: index of the calculated threshold (some value within Thr index)
+
+Procedure:
+- for each of 100 thresholds between min and max, call peaks with `peak_local_max`
```

Comment:
interesting. i wonder how this procedure compares to simple otsu-thresholding (http://scikit-image.org/docs/dev/auto_examples/xx_applications/plot_thresholding.html) could be something worth asking simone about